src/server/stream_family.cc

```diff
@@ -1379,6 +1414,8 @@ OpStatus OpSetId2(const OpArgs& op_args, string_view key, const streamID& sid) {
   if (!res_it)
     return res_it.status();
 
+  StreamMemTracker mem_tracker;
```

Comment:
Would it make sense to "attach" such memory trackers per object, and let the d'tor do the size diff calculation?

Reply:
The problem with the trackers is that it's not always possible to do the memory calculation at the end of the function (the destructor). This was mostly apparent with `json` because some commands would add temporary allocations inbetween which would affect the tracking accuracy. To mitigate this effect, we needed precise control for polling the difference on the heap, pretty much similarly to what I do here.

However, this is not really needed in a lot of places, especially for Streams (they have way less temporary waste mostly because they don't involve the non standard allocator behaviour we observed with json). 

Therefore, I will do a quick revamp and add a destructor.

Lastly, my plan is, once we get good coverage for all `objects` (meaning here all data types that we don't properly track and require a MemTracker) I would like to abstract them all under a `single interface` such that we don't get `JsonMemTracker`, `StreamMemTracker` etc but only `MemoryTracker` which will satisfy all the requirements for all the data types. The reason I haven't introduced such a generic interface yet is simply because I can't forsee all the requirements the types we don't properly track (my two cents would be that mostly we need either destructor tracking or manual tracking via a function).

Long rant, just thinking out loud of the long term goal here and the cleanness of the codebase

Reply:
Also another downside that comes to my mind, is that we might not want to track at all (let's say because of an early return or an error). In that case, we would need to `disable the tracker when it destructs` which bloats the code far more than a single function

Reply:
Sure, makes sense.